controllers/helper_test.go

```diff
@@ -538,6 +540,52 @@ func testBuildManagedZone(name, ns, domainName string) *kuadrantdnsv1alpha1.Mana
 	}
 }
 
+// testEndpointsTraversable consumes an array of endpoints and returns a boolean
+// indicating presence of that path from host to all IPs
+func testEndpointsTraversable(endpoints []*endpoint.Endpoint, host string, ips []string) bool {
+	for _, ip := range ips {
+		allTargetsFound := false
+		for _, ep := range endpoints {
+			// the target on the endpoint exists as an endpoint
+			if ep.DNSName == host {
+				// target is our IP
+				if slices.Contains(ep.Targets, ip) {
+					return true
```

Comment:
Why does this return here?

Reply:
this is the match for the IP in questions. Means we found an A record that contains the target IP and the fact that we got to the record means there is a path. We should return here to the higher levels. 
Although you led me to an error - if the second IP is not found we have already returned from the highest level of the recursion: fale positive change. Changed